src/main/java/org/jenkinsci/plugins/workflow/cps/view/InterpolatedSecretsAction.java

```diff
@@ -0,0 +1,163 @@
+    }
+
+    public void record(@Nonnull String stepName, @Nonnull List<String> interpolatedVariables, @Nonnull String nodeId) {
+        interpolatedWarnings.add(new InterpolatedWarnings(stepName, interpolatedVariables, run, nodeId));
+    }
+
+    public List<InterpolatedWarnings> getWarnings() {
+       return interpolatedWarnings;
+    }
+
+    public boolean hasWarnings() {
+        if (interpolatedWarnings.isEmpty()) {
+            return false;
+        } else {
+            return true;
+        }
+    }
+
+    public boolean isInProgress() {
+        return run.isBuilding();
+    }
+
+    @Override
+    public void onAttached(Run<?, ?> run) {
+        this.run = run;
+    }
+
+    @Override
+    public void onLoad(Run<?, ?> run) {
+        this.run = run;
+    }
+
+    @ExportedBean
+    public static class InterpolatedWarnings {
+        final String stepName;
+        final List<String> interpolatedVariables;
+        final Run run;
+        final String nodeId;
+
+        private InterpolatedWarnings(@Nonnull String stepName, @Nonnull List<String> interpolatedVariables, @Nonnull Run run, @Nonnull String nodeId) {
+            this.stepName = stepName;
+            this.interpolatedVariables = interpolatedVariables;
+            this.run = run;
+            this.nodeId = nodeId;
+        }
+
+        @Exported
+        public String getStepSignature() {
+            StringBuilder sb = new StringBuilder();
+            Map<String, Object> stepArguments;
+            try {
+                stepArguments = getStepArguments(run, nodeId);
+            } catch (IllegalStateException e) {
+                return "Unable to construct " +  stepName + ": " + e.getMessage();
+            }
+
+            sb.append(stepName + "(");
+            Set<Map.Entry<String, Object>> entrySet = stepArguments.entrySet();
+            if (!entrySet.isEmpty()) {
+                boolean first = true;
+                for (Map.Entry<String, Object> argEntry : stepArguments.entrySet()) {
+                    Object value = argEntry.getValue();
+                    String valueString = String.valueOf(value);
+                    if (value instanceof ArgumentsAction.NotStoredReason) {
+                        switch ((ArgumentsAction.NotStoredReason) value) {
+                            case OVERSIZE_VALUE:
+                                valueString = "argument omitted due to length";
+                                break;
+                            case UNSERIALIZABLE:
+                                valueString = "unable to serialize argument";
+                                break;
+                            default:
+                                break;
+                        }
+                    }
+                    if (first) {
+                        first = false;
+                    } else {
+                        sb.append(", ");
+                    }
+                    sb.append(argEntry.getKey() + ": " + valueString);
```

Comment:
I noticed  in  `UninstantiatedDescribable.toString()` that the arguments are printed out:  `firstArg=value` whereas I'm doing it here `firstArg: value`. Should I be matching the `UninstantiatedDescribable` format? I was just trying to make the step signature look more like what I have seen developers do in their Jenkinsfiles.

Reply:
`UninstantiatedDescribable.toString()` is for use from Java debuggers and the like so I doubt it is relevant to something you want to be `@Exported`.

Reply:
added parsing of `UninstantiatedDescribable` in 86cf9f3b961265edbeaef8243e10fc58ef4d1c78